app/Main.hs

```diff
@@ -2,46 +2,19 @@
 
 module Main where
 
-import           Data.Text           (Text, pack)
-import           Data.Text.IO        (readFile, writeFile)
-import           Prelude             hiding (readFile, writeFile)
-import           System.Environment  (getArgs)
-import           System.Exit
-import           System.IO           (hPutStrLn, stderr)
-import           Update.Nix.FetchGit
-import           Update.Nix.FetchGit.Utils
-import           Update.Nix.FetchGit.Warning
-import           Update.Span
+import qualified Data.Text
+import qualified System.Environment
+import qualified System.Exit
+import qualified Update.Nix.FetchGit
+
 
 main :: IO ()
 main =
   -- Super simple command line parsing at the moment, just look for one
-  -- filename.
-  getArgs >>= \case
-    [filename] -> processFile filename []
-    (filename:args) -> processFile filename (map pack args)
```

Comment:
lol, I wrote some pretty big-brained code here.

Reply:
:D I've altered that when adding `args` support.

Reply:
Also decided against `optparse-applicative` for now as it's dead simple but we might want to support different fetchers where it could come handy (I've noticed such PR for another project but don't quite remember which now). Ideally we would use native Haskell fetchers though :)